pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,2126 @@
+        :type voltage_range: :class:`.VoltageRange`, optional
+
+        """
+        voltage_range = VoltageRange(voltage_range)
+        enable = map_values(enable, {True: 1, False: 2})
+        internal_measurement = map_values(internal_measurement, {True: 1, False: 2})
+
+        self.write(f'rv {{ch}},{enable},{internal_measurement},{voltage_range.value}')
+
+    def voltage_source(self, source_range, source_value, current_compliance):
+        """ Sets the source range, source value and the current compliance
+        for the DC (constant voltage) measurement (``DV``).
+
+        :param source_range: Specifying source range
+        :type source_range: :class:`.VoltageRange`
+        :param source_value: A number specifying the source voltage value
+        :type source_value: float
+        :param current_compliance: A number specifying the current compliance
+        :type current_compliance: float
+
+        .. note::
+
+            Regardless of the specified current compliance polarity, both polarities (+ and -) are
+            set.
+            The current compliance range is automatically set to the minimum range that includes the
+            set value.
+
+        """
+        source_range = VoltageRange(source_range)
+        source_value = truncated_range(source_value, self.voltage_range)
+
+        self.write(f'dv {{ch}},{source_range.value},{source_value:.4e},{current_compliance:.4e}')
+
+    def voltage_pulsed_source(self, source_range, pulse_value, base_value, current_compliance):
+        """ Sets the source range, pulse value, base value and the current compliance
+        of the pulse (voltage) measurement (``PV``).
+
+        .. note::
+
+            Regardless of the specified current compliance polarity, both polarities (+ and -) are
+            set.
+            The current compliance range is automatically set to the minimum range that includes the
+            set value.
+
+        """
+        source_range = VoltageRange(source_range)
+        pulse_value = truncated_range(pulse_value, self.voltage_range)
+        base_value = truncated_range(base_value, self.voltage_range)
+
+        self.write(f'pv {{ch}},{source_range.value},{pulse_value:.4e},{base_value:.4e},'
+                   '{current_compliance:.4e}')
+
+    voltage_change_source = Channel.setting(
+        "spot {ch},%.4e",
+        """ Set target voltage from the current set voltage value (``SPOT``).
+
+        :type: float
+
+        .. note::
+
+            Only the DC action source value and pulse action pulse value
+            are changed using the currently set DC action and pulse action parameters.
+            Measure after the change and set the channel to output the measured data
+            to the specified ch. In other words, it's the same as running the following
+            commands:
+
+              1. DV/DI/PV/PI
+              2. XE xx
+              3. FCH xx
+
+        """,
+        cast=float,
```

Comment:
cast is used on the data returned by the device, not the data sent to the device. As you have a `setting`, nothing is returned and you do not need to specify a cast.
Additionally, float would be the default cast.